src/duckdb.rs

```diff
@@ -251,14 +251,42 @@ impl DuckDBTableProviderFactory {
     }
 
     pub async fn get_or_init_memory_instance(&self) -> Result<DuckDbConnectionPool> {
+        self.get_or_init_memory_instance_with_size_config(None, None)
+            .await
+    }
+
+    pub async fn get_or_init_file_instance(
+        &self,
+        db_path: impl Into<Arc<str>>,
+    ) -> Result<DuckDbConnectionPool> {
+        self.get_or_init_file_instance_with_size_config(db_path, None, None)
+            .await
+    }
+
+    pub async fn get_or_init_memory_instance_with_size_config(
```

Comment:
```suggestion
    pub async fn get_or_init_instance_with_builder(
```

This can just take the builder directly - the user constructs the builder they want to use, and then we don't need two separate `get_or_init` functions for memory or file - because its encoded in the builder already.